urlb879d9e865_ogotalski_elevator_tgz-pJ8-ElevatorTaskJ8/src/by/epam/lab/controller/ActionViewController.java

```diff
@@ -0,0 +1,179 @@
+					INPUT_ERROR, JOptionPane.ERROR_MESSAGE);
+		}
+	}
+
+	private void drawLogPanel() {
+		JTextArea logArea = new JTextArea();
+
+		JScrollPane scrollPane = new JScrollPane(logArea);
+		scrollPane.setPreferredSize(new Dimension(Integer.MAX_VALUE, 200));
+		scrollPane.setMinimumSize(new Dimension(Integer.MAX_VALUE, 200));
+		mainFrame.addComponent(scrollPane);
+		mainFrame.repaint();
+		LOG.addAppender(new TextAreaAppender(logArea));
+	}
+
+	private void getParamsFromControlPanel() {
+		Configuration configuration = Configuration.getConfiguration();
+		configuration.setElevatorCapacity(Integer.parseInt(controlPanel.getElevatorCapacity()));
+		configuration.setPassengersNumber(Integer.parseInt(controlPanel.getPassengersNumber()));
+		configuration.setStoriesNumber(Integer.parseInt(controlPanel.getStoriesNumber()));
+		configuration.setAnimationBoost(controlPanel.getAnimationBoost());
+		
+	}
+
+	private void drawFloorPanel() {
+		floorsMap = new HashMap<Floor, FloorView>(Configuration.getConfiguration().getStoriesNumber());
+		JPanel panel = new JPanel();
+		panel.setLayout(new BoxLayout(panel, BoxLayout.Y_AXIS));
+		FloorView.setNumPassengersView(building.getElevator().getCapacity());
+		fillFloorsMap(panel);
+		JScrollPane pane = new JScrollPane(panel);
+		mainFrame.addComponent(pane);
+		timer = new Timer(Configuration.getConfiguration().getSleepTime(),
+				new UpdateListener(this));
+		timer.start();
+	}
+
+	private void fillFloorsMap(JPanel panel) {
+		FloorView view;
+		List<Floor> floors = building.getFloors();
+		Elevator elevator = building.getElevator();
+		ReverseIterator<Floor> iterator = new ReverseIterator<Floor>(floors);
+		Floor floor;
+		while (iterator.hasNext()) {
+			floor = iterator.next();
+			view = new FloorView(floor.getDispatchPassengers(),
+					floor.getArrivalPassengers());
+			panel.add(view);
+			floorsMap.put(floor, view);
+		}
+		view = floorsMap.get(elevator.getCurrentFloor());
+		view.setElevatorPassengers(elevator.getElevatorPassengers());
+		panel.setVisible(true);
+	}
+
+	public void updateView() {
+
+		for (Floor floor : building.getFloors()) {
+			floorsMap.get(floor).update(floor.getDispatchPassengers(),
+					floor.getArrivalPassengers());
+
+		}
+		Elevator elevator = building.getElevator();
+		floorsMap.get(elevator.getCurrentFloor()).setElevatorPassengers(
+				elevator.getElevatorPassengers());
+		mainFrame.repaint();
+	}
+
+	@Override
+	public void abort() {
+		LOG.info(ABORTING_TRANSPORTATION);
+		threadGroup.interrupt();
+		timer.stop();
+	}
+
+	@Override
+	public void viewLog() {
+		new LogViewFrame();
+
+	}
+
+	
+
+	@Override
+	public void run() {
+		try {
+			building.startElevator(Configuration.getConfiguration().getSleepTime());
+			if (building.verify()) {
+				JOptionPane.showMessageDialog(mainFrame, VALIDATION_COMPLETED);
+			} else {
+				JOptionPane.showMessageDialog(mainFrame, VALIDATION_ERROR,
+						VALIDATION, JOptionPane.ERROR_MESSAGE);
+			}
+		} catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)